File: app/src/main/java/com/huanchengfly/tieba/post/components/MyViewHolder.java
package com.huanchengfly.tieba.post.components;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.IdRes;
import androidx.annotation.LayoutRes;
import androidx.annotation.StringRes;
import androidx.recyclerview.widget.RecyclerView;

public class MyViewHolder extends RecyclerView.ViewHolder {
    private Context mContext;

    public MyViewHolder(View itemView) {
        super(itemView);
        this.mContext = itemView.getContext();
    }

    public MyViewHolder(Context context, @LayoutRes int layoutId) {
        super(View.inflate(context, layoutId, null));
        this.mContext = context;
    }

    public <T extends View> T getView(@IdRes int id) {
        return itemView.findViewById(id);
    }

    public void setItemOnClickListener(View.OnClickListener listener) {
        itemView.setOnClickListener(listener);
    }

    public void setItemOnLongClickListener(View.OnLongClickListener listener) {
        itemView.setOnLongClickListener(listener);
    }

    public void setOnClickListener(int id, View.OnClickListener onClickListener) {
        View view = getView(id);
        if (view != null) {
            view.setOnClickListener(onClickListener);
        }
    }

    public void setText(@IdRes int id, CharSequence text) {
        View view = getView(id);
        if (view instanceof TextView) {
            ((TextView) view).setText(text);
        }
    }

    public void setVisibility(int id, int visibility) {
        View view = getView(id);
        if (view != null) {
            view.setVisibility(visibility);
        }
    }

    public void setText(@IdRes int id, @StringRes int string) {
        setText(id, this.mContext.getString(string));
    }
}

File: app/src/main/java/com/huanchengfly/tieba/post/components/MyViewHolder.kt
package com.huanchengfly.tieba.post.components

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnLongClickListener
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.DrawableRes
import androidx.annotation.IdRes
import androidx.annotation.LayoutRes
import androidx.annotation.StringRes
import androidx.recyclerview.widget.RecyclerView

class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val mContext: Context = itemView.context

    @JvmOverloads
    constructor(context: Context, @LayoutRes layoutId: Int, parent: ViewGroup? = null) : this(
        LayoutInflater.from(context).inflate(layoutId, parent, false)
    )

    fun <T : View> getView(@IdRes id: Int): T {
        return itemView.findViewById(id)
    }

    fun <T : View> getViewOrNull(@IdRes id: Int): T? {
        return itemView.findViewById(id)
    }

    fun setItemOnClickListener(listener: View.OnClickListener?) {
        itemView.setOnClickListener(listener)
    }

    fun setItemOnLongClickListener(listener: OnLongClickListener?) {
        itemView.setOnLongClickListener(listener)
    }

    fun setOnClickListener(id: Int, onClickListener: View.OnClickListener?) {
        val view = getView<View>(id)
        view.setOnClickListener(onClickListener)
    }

    fun setText(@IdRes id: Int, text: CharSequence?) {
        val view = getView<View>(id)
        if (view is TextView) {
            view.text = text
        }
    }

    fun setVisibility(id: Int, visibility: Int) {
        val view = getView<View>(id)
        view.visibility = visibility
    }

    @JvmOverloads
    fun setVisibility(id: Int, visibility: Boolean, gone: Boolean = true) {
        val view = getView<View>(id)
        view.visibility = if (visibility) View.VISIBLE else if (gone) View.GONE else View.INVISIBLE
    }

    fun setText(@IdRes id: Int, @StringRes string: Int) {
        setText(id, mContext.getString(string))
    }

    fun setImageDrawable(@IdRes id: Int, drawable: Drawable?) {
        getView<ImageView>(id).setImageDrawable(drawable)
    }

    fun setImageResource(@IdRes id: Int, @DrawableRes drawableRes: Int) {
        getView<ImageView>(id).setImageResource(drawableRes)
    }
}